apps/server/src/modules/tool/external-tool/mapper/external-tool-datasheet.mapper.ts

```diff
@@ -0,0 +1,174 @@
+				externalToolData.skipConsent = 'nein';
+			}
+		}
+
+		if (ExternalTool.isLti11Config(externalTool.config)) {
+			externalToolData.messageType = externalTool.config.lti_message_type;
+			externalToolData.privacy = externalTool.config.privacy_permission;
+		}
+
+		return externalToolData;
+	}
+
+	private static mapToInstanceName(): string {
+		if (Configuration.get('SC_THEME') === 'n21') {
+			return 'Niedersächsische Bildungscloud';
+		}
+
+		if (Configuration.get('SC_THEME') === 'brb') {
+			return 'Schul-Cloud Brandenburg';
+		}
+
+		if (Configuration.get('SC_THEME') === 'thr') {
+			return 'Thüringer Schulcloud';
+		}
+
+		if (Configuration.get('SC_THEME') === 'default') {
+			return 'dBildungscloud';
+		}
+
+		return 'unbekannt';
+	}
+
+	private static mapToLimitedContexts(externalTool: ExternalTool): string[] {
+		const restrictToContexts: string[] = [];
+		if (externalTool.restrictToContexts?.includes(ToolContextType.COURSE)) {
+			restrictToContexts.push('Kurs ');
+		}
+		if (externalTool.restrictToContexts?.includes(ToolContextType.BOARD_ELEMENT)) {
+			restrictToContexts.push('Kurs-Board');
+		}
+
+		return restrictToContexts;
+	}
+
+	private static mapToParameterDataList(externalTool: ExternalTool): ExternalToolParameterDatasheetTemplateData[] {
+		const parameterData: ExternalToolParameterDatasheetTemplateData[] = [];
+
+		externalTool.parameters?.forEach((parameter: CustomParameter) => {
+			const paramData: ExternalToolParameterDatasheetTemplateData =
+				ExternalToolDatasheetMapper.mapToParameterData(parameter);
+			parameterData.push(paramData);
+		});
+
+		return parameterData;
+	}
+
+	private static mapToParameterData(parameter: CustomParameter): ExternalToolParameterDatasheetTemplateData {
+		const parameterData: ExternalToolParameterDatasheetTemplateData = new ExternalToolParameterDatasheetTemplateData({
+			name: parameter.name,
+			type: ExternalToolDatasheetMapper.mapToType(parameter),
+			properties: ExternalToolDatasheetMapper.mapToProperties(parameter),
+			scope: ExternalToolDatasheetMapper.mapToScope(parameter),
+			location: parameter.location,
+		});
+
+		return parameterData;
+	}
+
+	private static mapToProperties(parameter: CustomParameter): string {
+		const properties: ExternalToolParameterDatasheetTemplateProperty[] = [];
+		let propertiesString = '';
+		if (parameter.isOptional) {
+			properties.push(ExternalToolParameterDatasheetTemplateProperty.OPTIONAL);
+		}
+
+		if (parameter.isProtected) {
+			properties.push(ExternalToolParameterDatasheetTemplateProperty.PROTECTED);
+		}
+
+		properties.forEach((property: ExternalToolParameterDatasheetTemplateProperty) => {
+			propertiesString += `${property}, `;
+		});
+
+		propertiesString = propertiesString.substring(0, propertiesString.length - 2);
+		return propertiesString;
+	}
+
+	private static mapToType(parameter: CustomParameter): string {
+		let type = '';
+		switch (parameter.type) {
+			case CustomParameterType.STRING:
+				type = 'Zeichenkette';
```

Comment:
I prefer using a direct return instead of a break, when your function is only a switch.